File: src/main/java/com/example/eventsourcing/config/EventSourcingProperties.java
package com.example.eventsourcing.config;

import com.example.eventsourcing.domain.AggregateType;

import java.util.Map;

public class EventSourcingProperties {

    private static final SnapshottingProperties NO_SNAPSHOTTING = new SnapshottingProperties(false, 0);

    private final Map<AggregateType, SnapshottingProperties> snapshotting;

    public EventSourcingProperties(Map<AggregateType, SnapshottingProperties> snapshotting) {
        this.snapshotting = snapshotting;
    }

    public SnapshottingProperties getSnapshotting(AggregateType aggregateType) {
        return snapshotting.getOrDefault(aggregateType, NO_SNAPSHOTTING);
    }

    public record SnapshottingProperties(
            boolean enabled,
            int nthEvent
    ) {
    }
}

File: src/main/kotlin/com/example/eventsourcing/config/EventSourcingProperties.kt
package com.example.eventsourcing.config

import com.example.eventsourcing.config.SnapshottingProperties.Companion.NO_SNAPSHOTTING
import com.example.eventsourcing.domain.model.AggregateType

class EventSourcingProperties(private val snapshotting: Map<AggregateType, SnapshottingProperties>) {
    fun getSnapshotting(aggregateType: AggregateType): SnapshottingProperties {
        return snapshotting[aggregateType] ?: NO_SNAPSHOTTING
    }
}
